src/components/dataTable/js/tableCellDirective.js

```diff
@@ -0,0 +1,213 @@
+  var headRows = [].slice.call(this.table.getHead().rows);
+  var cellIndex = this.$element[0].cellIndex;
+
+  return headRows
+    .map(function(row) {
+      return angular.element(row.cells[cellIndex]).data('align');
+    })
+    .filter(function(cell) {
+      return !!cell;
+    })[0];
+};
+
+function MdTableAlignDirective() {
+
+  return  {
+    restrict: 'A',
+    require: ['mdCell', '^^mdHead'],
+    link: postLink
+  };
+
+  function postLink(scope, element, attrs) {
+    attrs.$observe('mdAlign', function(align) {
+      element.data('align', align);
+    });
+  }
+}
+
+function MdTableOrderByDirective() {
+
+  return {
+    restrict: 'A',
+    require: ['mdOrderBy', 'mdCell', '^^mdOrder', '^^mdTable'],
+    scope: {},
+    controller: MdTableOrderByController,
+    controllerAs: '$mdOrderBy',
+    link: postLink
+  };
+
+  function postLink(scope, element, attrs, ctrls) {
+    var ctrl = ctrls[0];
+    var cellCtrl = ctrls[1];
+    var orderCtrl = ctrls[2];
+    var tableCtrl = ctrls[3];
+
+    ctrl.initCtrl(tableCtrl, orderCtrl, cellCtrl);
+  }
+}
+
+function MdTableOrderByController($element, $scope, $attrs, $compile, $mdUtil, $$mdSvgRegistry) {
+  this.$element = $element;
+  this.$scope = $scope;
+  this.$attrs = $attrs;
+  this.$compile = $compile;
+  this.$mdUtil = $mdUtil;
+  this.$$mdArrowUpIcon = $$mdSvgRegistry.mdArrowUp;
+
+  this.isActiveRegex = new RegExp('^-?' + this.$attrs.mdOrderBy + '$');
+
+  this.$attrs.$observe('mdOrderBy', function (orderBy) {
+    if (orderBy) {
+      this.enableSorting();
+    } else {
+      this.disableSorting();
+    }
+  }.bind(this));
+}
+
+MdTableOrderByController.prototype.initCtrl = function(tableCtrl, orderCtrl, cellCtrl) {
+  this.table = tableCtrl;
+  this.orderCtrl = orderCtrl;
+  this.cellCtrl = cellCtrl;
+
+  this.unregisterWatch = null;
+  this.clickListenerFn = this.updateSortOrder.bind(this);
+
+  // Watch for position changes to realign the sort icon.
+  this.$scope.$watch(function() {
+    return this.cellCtrl.$element[0].cellIndex;
+  }.bind(this), function() {
+    var icon = this.$element[0].querySelector('.md-sort-icon');
+
+    if (icon) {
+      if (this.$element.data('align') === 'right') {
+        this.$element.prepend(icon);
+      } else {
+        this.$element.append(icon);
+      }
+    }
+
+  }.bind(this));
+};
+
+MdTableOrderByController.prototype.isActive = function() {
+  // The isActiveRegex checks the headCtrl's order property.
+  // When the order direction is inverted, we prefix the order property
+  // with a minus.
+  return this.isActiveRegex.test(this.orderCtrl.value);
+};
+
+MdTableOrderByController.prototype.getDirection = function() {
+  if (this.isActive()) {
+    // When the current order value is prefixed with a minus, then its order
+    // is descendant.
+    return this.orderCtrl.value.charAt(0) === '-' ? 'md-desc' : 'md-asc';
+  } else {
+    return this.$mdUtil.parseAttributeBoolean(this.$attrs.mdDesc) ? 'md-desc' : 'md-asc';
+  }
+};
```

Comment:
For sure it can. But sometimes we explicitly use a normal `if`, because it allows us to add comments and not exceed the 100 lines limit.

It's just clear and reducing to a minimum with a ternary operator is not always the best solution.